### ISSUE TITLE
skaffold says all is well even when there were validation errors

### ISSUE THREAD

**Issue by andsens** [GoogleContainerTools/skaffold#1738]
While examining the error from #1737 I noticed that skaffold said all was fine and that the deployment completed, even though the manifest failed to validate.

### Expected behavior

skaffold returns with a non-zero exit code when manifest validation fails.

### Actual behavior

skaffold return with 0 and actually says "Deploy complete"

### Information

- Skaffold version: 0.24.0
- Operating system: MacOS 10.14.3
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta6
kind: Config
deploy:
  kustomize:
    path: .
```

### Steps to reproduce the behavior

1. `skaffold run` [this gist](https://gist.github.com/andsens/19a7485b753bf303fb3a9d768e457ee4).

### Full log

```
$ skaffold run
Starting build...
Build complete in 40.014µs
Starting test...
Test complete in 7.814µs
Starting deploy...
kubectl client version: 1.13
error: error validating "STDIN": error validating data: ValidationError(CustomResourceDefinition.spec.validation.openAPIV3Schema.properties.metadata): unknown field "labels" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1beta1.JSONSchemaProps; if you choose to ignore these errors, turn validation off with --validate=false
Deploy complete in 644.45351ms
You can also run [skaffold run --tail] to get the logs
```

**Comment by tejal29** [comment]
Thanks @andsens for this bug report.  Just to confirm,  `kubectl deploy` end up deploying any other resources?

**Comment by andsens** [comment]
The log is from the manifest in the gist, which only contains the one resource. I haven't actually checked what happens when the manifests partially validate. Let me check on Friday and I'll report back.

**Comment by tejal29** [comment]
no problem!

**Comment by andsens** [comment]
@tejal29 no other resources seem to deploy. I tested by adding this pod.yaml
```
---
apiVersion: v1
kind: Pod
metadata:
  name: busybox
spec:
  containers:
  - name: busybox
    image: busybox:1.28
    command:
    - sleep
    - "3600"
    imagePullPolicy: IfNotPresent
  restartPolicy: Always
```

I tried adding `pod.yaml` both before and after `crd.yaml` in `kustomization.yaml` with the same result. The output was identical to the one above.

When removing `crd.yaml` from `kustomization.yaml` the pod deploys as expected of course :-)

**Comment by tejal29** [comment]
Thanks @andsens. Someone from the team will pick it up once triaged.

**Comment by daniyalj** [comment]
running into the same issue. Im deploying rbac-manager with skaffold with the following:

```
$ cat skaffold.yaml 
apiVersion: skaffold/v1beta8
kind: Config
deploy:
  kubectl:
    manifests:
      - manifests/rbac-manager/01-rbac-manager.yaml
```

01-rbac-manager.yaml is [here](https://github.com/reactiveops/rbac-manager/blob/master/deploy/all.yaml)

error is:
```
clusterrole.rbac.authorization.k8s.io/rbac-manager unchanged
clusterrolebinding.rbac.authorization.k8s.io/rbac-manager unchanged
error: error validating "STDIN": error validating data: ValidationError(CustomResourceDefinition.spec.validation.openAPIV3Schema.properties.metadata): unknown field "labels" in io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1beta1.JSONSchemaProps; if you choose to ignore these errors, turn validation off with --validate=false
FATA[0000] deploy failed: kubectl apply: exit status 1 
```

**Comment by nkubala** [comment]
closed via #2076

**Comment by andsens** [comment]
Awesome stuff. Thanks @pyaillet!
